velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -0,0 +1,267 @@
+      veloxType, std::move(params), veloxFunction);
+}
+
+std::shared_ptr<const core::ConstantTypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::Literal& sLit) {
+  auto typeCase = sLit.literal_type_case();
+  switch (typeCase) {
+    case ::substrait::Expression_Literal::LiteralTypeCase::kFp64: {
+      return std::make_shared<core::ConstantTypedExpr>(sLit.fp64());
+    }
+    case ::substrait::Expression_Literal::LiteralTypeCase::kBoolean: {
+      return std::make_shared<core::ConstantTypedExpr>(sLit.boolean());
+    }
+    default:
+      VELOX_NYI(
+          "Substrait conversion not supported for type case '{}'", typeCase);
+  }
+}
+
+std::shared_ptr<const core::ITypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression& sExpr,
+    const int32_t& inputPlanNodeId) {
+  std::shared_ptr<const core::ITypedExpr> veloxExpr;
+  auto typeCase = sExpr.rex_type_case();
+  switch (typeCase) {
+    case ::substrait::Expression::RexTypeCase::kLiteral: {
+      veloxExpr = toVeloxExpr(sExpr.literal());
+      break;
+    }
+    case ::substrait::Expression::RexTypeCase::kScalarFunction: {
+      veloxExpr = toVeloxExpr(sExpr.scalar_function(), inputPlanNodeId);
+      break;
+    }
+    case ::substrait::Expression::RexTypeCase::kSelection: {
+      veloxExpr = toVeloxExpr(sExpr.selection(), inputPlanNodeId);
+      break;
+    }
+    default:
+      VELOX_NYI(
+          "Substrait conversion not supported for Expression '{}'", typeCase);
+  }
+  return veloxExpr;
+}
+
+// TODO: Support different types here.
+class SubstraitVeloxExprConverter::FilterInfo {
+ public:
+  FilterInfo() {}
+  // Used to set the left bound.
+  void setLeft(double left, bool isExclusive) {
+    left_ = left;
+    leftExclusive_ = isExclusive;
+    if (!isInitialized_) {
+      isInitialized_ = true;
+    }
+  }
+
+  // Used to set the right bound.
+  void setRight(double right, bool isExclusive) {
+    right_ = right;
+    rightExclusive_ = isExclusive;
+    if (!isInitialized_) {
+      isInitialized_ = true;
+    }
+  }
+
+  // Will fordis Null value if called once.
+  void forbidsNull() {
+    nullAllowed_ = false;
+    if (!isInitialized_) {
+      isInitialized_ = true;
+    }
+  }
+
+  // Return the initialization status.
+  bool isInitialized() {
+    return isInitialized_ ? true : false;
+  }
+
+  // The left bound.
+  std::optional<double> left_ = std::nullopt;
+  // The right bound.
+  std::optional<double> right_ = std::nullopt;
+  // The Null allowing.
+  bool nullAllowed_ = true;
+  // If true, left bound will be exclusive.
+  bool leftExclusive_ = false;
+  // If true, right bound will be exclusive.
+  bool rightExclusive_ = false;
+
+ private:
+  bool isInitialized_ = false;
+};
+
+void SubstraitVeloxExprConverter::getFlatConditions(
+    const ::substrait::Expression& sFilter,
+    std::vector<::substrait::Expression_ScalarFunction>* scalarFunctions) {
```

Comment:
pointers are usually to specify this parameter is optional, but it doesn't seem to be the case. Could we take it as a non-const ref instead?